server/routes/csip-records/view.njk

```diff
@@ -1,74 +1,130 @@
 {% from "govuk/components/button/macro.njk" import govukButton %}
-{% from "moj/components/page-header-actions/macro.njk" import mojPageHeaderActions %}
 {%- from "moj/components/sub-navigation/macro.njk" import mojSubNavigation -%}
 {% from "govuk/components/notification-banner/macro.njk" import govukNotificationBanner %}
 {% from "partials/statusTag/macro.njk" import statusTag %}
 {% from "govuk/components/warning-text/macro.njk" import govukWarningText %}
+{%- from "moj/components/button-menu/macro.njk" import mojButtonMenu -%}
 
 {% set fullWidth = true %}
 
 {% block additionalScripts %}
   <script type="text/javascript" nonce="{{ cspNonce }}">
-      const printButton = document.getElementById('print-button');
-      if (printButton === null) {
-        if (!window.__customPrintObserver) {
-          const observer = new MutationObserver(mutationList => {
-            mutationList.forEach(mutation => {
-              if (mutation.type === 'childList') {
-                mutation.addedNodes.forEach(node => {
-                  if (node.id === 'print-button' && node.classList.contains('govuk-!-display-none')) {
-                    node.classList.remove('govuk-!-display-none');
-                    node.ariaHidden = false;
-                    node.onclick = function () { print(); }
-                  }
-                });
-              }
-            })
-          });
-
-          observer.observe(document.documentElement, { attributes: false, childList: true, subtree: true })
-
-          window.__customPrintObserver = observer
+    const printOptionsButton = document.getElementById('print-options-button');
+    if (printOptionsButton) {
+      printOptionsButton.classList.remove('govuk-!-display-none')
+    }
+
+    const togglePrintBlock = (id, enabled, addPageBreak = false) => {
+      const block = document.getElementById(id)
+      if (block) {
+        if (enabled) {
+          block.classList.add('print-hidden-block')
+          if (addPageBreak) {
+            block.classList.add('page-break-print')
+          } else {
+            block.classList.remove('page-break-print')
+          }
+        } else {
+          block.classList.remove('print-hidden-block')
+          block.classList.add('govuk-!-display-none-print')
         }
-      } else {
-        printButton.classList.remove('govuk-!-display-none');
-        printButton.removeAttribute('aria-hidden');
-        printButton.onclick = function () { print(); }
       }
+    }
+
+    const printAllButton = document.getElementById('print-all-button')
+    if (printAllButton) {
+      printAllButton.onclick = () => {
+        togglePrintBlock('csip-referral', true, false)
```

Comment:
Perhaps to make it clearer what the result of each of these calls are doing, there could be a displayForPrint and hideFromPrint method that calls the underlying togglePrintBlock function, Also wonder if that could return a function pointer for displayForPrint('csip-investigation').andAddPageBreak() or just have displayForPrint and displayForPrintAndAddPageBreak separately. hideFromPrint could also take an array. Then you end up with just

displayForPrint('csip-referral');
hideFromPrint(['csip-investigation', 'csip-plan', 'csip-reviews'])

For the print referral onclick for example